### PR TITLE
ARTEMIS-4238 deprecate transactionTimeout activation config prop

### PR DESCRIPTION
**Pull request by jbertram**

### CODE REVIEW

File: artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationSpec.java [hunk 1]
```diff
@@ -120,6 +120,7 @@ public class ActiveMQActivationSpec extends ConnectionFactoryProperties implemen
    /**
     * Transaction timeout
     */
+   @Deprecated
```
Comment (gemmellr) [inline artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationSpec.java:123]: If it is deprecated for future removal as indicated in the documentation, should the Deprecated tag indicate that with the forRemoval flag?

File: artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationSpec.java [hunk 2]
```diff
@@ -631,6 +632,7 @@ public Integer getTransactionTimeout() {
     *
     * @param value The value
     */
+   @Deprecated
```
Comment (gemmellr) [inline artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationSpec.java:635]: Ditto

File: artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationSpec.java (hunk 1 above)
Comment (jbertram) [reply artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationSpec.java]: Done.

File: artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationSpec.java (hunk 2 above)
Comment (jbertram) [reply artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationSpec.java]: Ditto

**Pull request merged**